src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions_CreateEqualsMethod.cs

```diff
@@ -150,18 +157,19 @@ public static IMethodSymbol CreateEqualsMethod(this Compilation compilation, Imm
                 //
                 //      myType != null
                 expressions.Add(factory.ReferenceNotEqualsExpression(localNameExpression, factory.NullLiteralExpression()));
-                if (HasExistingBaseEqualsMethod(containingType, cancellationToken))
-                {
-                    // If we're overriding something that also provided an overridden 'Equals',
-                    // then ensure the base type thinks it is equals as well.
-                    //
-                    //      base.Equals(obj)
-                    expressions.Add(factory.InvocationExpression(
-                        factory.MemberAccessExpression(
-                            factory.BaseExpression(),
-                            factory.IdentifierName(EqualsName)),
-                        objNameExpression));
-                }
+            }
+
+            if (!containingType.IsValueType && HasExistingBaseEqualsMethod(containingType, cancellationToken))
```

Comment:
>!containingType.IsValueType && [](start = 16, length = 30)

I didn't understand this change (`!containingType.IsValueType &&`)

Reply:
so, previously this was just in the "else" branch.  That was fine when this code only did "is this a value type, if so, generate one way, otherwise generate the other".  Now we have 3 cases:

1. is this a pattern (value or reference type)
2. is this a value type.
3. otherwise...

only in case '2' do we not want to call 'base.Equals' i.e. value types never want to call base.Equals as that's always bad and expensive to call the base, reflectoin based, System.ValueType.Equals.  Case '1' and '3' want to call base.Equals if there is a base Equals method.

Does that make sense? 

Note: this is tested.